src/content/manifest/okteto-variables.mdx

```diff
@@ -43,6 +64,127 @@ To delete an existing variable, click on the **Delete** button on the right. You
+
+- **`OKTETO_DOMAIN`**: The domain where Okteto exposes your application endpoints.
+- **`OKTETO_NAMESPACE`**: The namespace where your application is installed.
+- **`OKTETO_USERNAME`**: Your username in Okteto.
+- **`OKTETO_REGISTRY_URL`**: The URL of the Okteto Registry.
+- **`OKTETO_GIT_BRANCH`**: The name of the Git branch being deployed.
+- **`OKTETO_GIT_COMMIT`**: The SHA1 hash of the last commit of the branch.
+
+### **Built-in Environment Variables for Images in Okteto Registry**
+
+Reference images defined in the `build` section of the Okteto manifest using the following environment variables:
+
+- **`OKTETO_BUILD_(IMAGE)_IMAGE`**: Full image reference.
+- **`OKTETO_BUILD_(IMAGE)_REGISTRY`**: Registry URL where the image was pushed.
+- **`OKTETO_BUILD_(IMAGE)_REPOSITORY`**: Name of the image that was pushed.
+- **`OKTETO_BUILD_(IMAGE)_SHA`**: Latest tag and SHA of the image.
+
+For instance, with a build section like:
+
+```yaml
+build:
+  hello-world:
+    context: .
+```
+
+Okteto will generate environment variables such as:
+
+- **`OKTETO_BUILD_HELLO_WORLD_IMAGE`**: registry.okteto.example.com/cindy/hello-world@sha256:xxx
+- **`OKTETO_BUILD_HELLO_WORLD_REGISTRY`**: registry.okteto.example.com
+- **`OKTETO_BUILD_HELLO_WORLD_REPOSITORY`**: cindy/hello-world
+- **`OKTETO_BUILD_HELLO_WORLD_SHA`**: f56119a37b@sha256:xxx
+
+> Note: If an `image` name includes '-', it will be replaced by '_' in the corresponding environment variables.
+
+### **Built-in Environment Variables for Dependencies:**
+Reference variables defined by your dependencies using the following pattern:
+
+- **`OKTETO_DEPENDENCY_<dependency>_VARIABLE_<variable>`**: The value of a variable of a dependency.
+    
+For example, if you have a dependency in your Okteto Manifest:
+
+```yaml
+dependencies:
+  database:
+      repository: https://github.com/okteto/my-database
+      variables:
+      USERNAME: okteto
+      wait: true
+```
+
+And the repository has this Okteto Manifest:
+
+```yaml
+deploy:
+  - echo "PASSWORD=1234" >> $OKTETO_ENV
+```
+
+Reference the variables in your Okteto Manifest deploy section like this:
+
+```yaml
+dependencies:
+  database:
+    repository: https://github.com/okteto/my-database
+    variables:
+    USERNAME: okteto
+    wait: true
+deploy:
+  - echo ${OKTETO_DEPENDENCY_DATABASE_VARIABLE_USERNAME}
+  - echo ${OKTETO_DEPENDENCY_DATABASE_VARIABLE_PASSWORD}
+```
+    
+> Note: USERNAME has a static value, and PASSWORD is dynamically generated.
+
+## Feature flag variables
+
+Okteto provides a list of variables that serve as feature flags for some of the features we release in our product. Below is a list of those variables that you can use to leverage these newer and experimental features:
+
+- `OKTETO_SMART_BUILDS_ENABLED`: enables the "smart builds" feature that uses a hash of the build context to determine if an image needs to be rebuilt or can be pulled from the registry. This can accelerate build times.
```

Comment:
@pchico83 Are there any other feature flag variables to include here?

Reply:
I'm supportive of documenting them now quickly and concisely. A follow-on iteration could expand further if we need to.

Is there an internal document or other resource to help me understand what each does so I can populate this content?

Reply:
@jLopezbarb could you help to document these variables?

Reply:
Resolving this to merge the PR since it's approved and all other notes are addressed.

Will submit a follow-up PR once we have this list with details 👍